js/customize-object-selector-control.js

```diff
@@ -414,6 +424,20 @@
 			var control = this;
 			control.actuallyEmbed();
 			api.Control.prototype.focus.call( control, args );
+		},
+
+		/**
+		 * Add featured image if select2 data is returning.
+		 *
+		 * @param {object} data post data.
+		 * @returns {string}
+		 */
+		select2Template: function(data){
+			if ( data.hasOwnProperty( 'featured_image' ) && data.featured_image.hasOwnProperty( 'sizes' ) && data.featured_image.sizes.hasOwnProperty( 'thumbnail' ) && data.featured_image.sizes.thumbnail.hasOwnProperty( 'url' ) && ! _.isEmpty( data.featured_image.sizes.thumbnail.url ) ) {
```

Comment:
I think it can be a bit less verbose, like:

```js
if ( data.featured_image && data.featured_image.sizes && data.featured_image.sizes.thumbnail ) {

}
```